APIServer/api_endpoints.py

```diff
@@ -116,23 +115,62 @@ def get(self):
         return registry.to_json()
 
 
+model_name_defn = api.model("model_name", {
+    "model_name": fields.String("Name of the model")
+})
+
+
 @api.route('/models/<int:exec_key>')
 class Model(Resource):
     """
     Read a single model from the registry.
     """
+
     @api.response(HTTP_SUCCESS, 'Success')
     @api.response(HTTP_NOT_FOUND, 'Not Found')
     def get(self, exec_key):
         model = get_model_if_exists(exec_key)
         return json_converter(model)
 
+    """
+    Setup a test model in registry
+    """
+
+    @api.response(HTTP_SUCCESS, 'Success')
+    @api.response(HTTP_NOT_FOUND, 'Not Found')
+    @api.expect(model_name_defn)
+    def post(self, exec_key):
```

Comment:
Hi @gcallah ,
We talked about adding a new api to setup test model at reserved location as a POST method for /'models/exec_key end point where exec_key is optional.
Since the resource is defined as `models/exec_key` I am not sure how to make the exec_key optional without converting it to a query param?